### DISCUSSION TITLE
Edge Handler: setPreviewData() for setting preview mode cookies missing

### DISCUSSION THREAD

**Discussion by styxlab** [vercel/vercel#6903] [Ideas]
Thanks so much for bringing edge functions with Next.js 12: that opens up so many new possibilities!

It would be great to have `setPreviewData()` on `NextResponse` from `next/server` in order to be able to set the preview cookie in the edge handler. Currently it is only available in `NextApiResponse`.

Maybe its already possible to set the cookie manually with the available `cookies()` object but I have not found a way to do this correctly.

**Comment by leerob** [comment]
I think this should already be possible by manually setting cookies, but agreed it could be nice to have a built-in way of doing this!
